src/tools/tidy/src/unit_tests.rs

```diff
@@ -1,44 +1,60 @@
 //! Tidy check to ensure `#[test]` and `#[bench]` are not used directly inside
-//! `core` or `alloc`.
+//! of the standard library.
 //!
 //! `core` and `alloc` cannot be tested directly due to duplicating lang items.
 //! All tests and benchmarks must be written externally in
 //! `{coretests,alloctests}/{tests,benches}`.
 //!
-//! Outside of `core` and `alloc`, tests and benchmarks should be outlined into
-//! separate files named `tests.rs` or `benches.rs`, or directories named
+//! Outside of the standard library, tests and benchmarks should be outlined
```

Comment:
Should this say "inside"? If not, I am very confused why this comment gives advice for how to write non-standard-library tests...

Also, what should I do if my test needs access to private functions or fields? Making them all unstably public seems pretty hacky.

Reply:
Outside of the standard library (so eg inside the compiler), tests should be in separate files. These should generally be submodules of the module that you want to test and thus have access to private functions and fields.

Inside the standard library, tests should be not just in separate files, but also in separate packages. Once all standard library tests have actually been moved to separate packages, we can get rid of all lib.miri.rs and testing will become less fragile.